Utils/azuretools-core/src/com/microsoft/azuretools/utils/WebAppUtils.java

```diff
@@ -371,60 +371,34 @@ public ResourceGroup getRg() {
     }
 
     /**
-     * app.linuxFxVersion() is the only API we could get the version info of a Linux web app.
-     * It returns values like "Tomcat|8.5-jre8" if it is a Linux with the web container Tomcat.
+     * Check if the web app is a Windows or Linux Java configured web app.
+     * Docker web apps are not included.
      */
-    public static String getJDKVersion(@NotNull final WebApp webApp) {
-        switch(webApp.operatingSystem()) {
-            case WINDOWS:
-                return webApp.javaVersion().toString();
-            case LINUX:
-                final String linuxVersion = webApp.linuxFxVersion();
-                if (linuxVersion == null) {
-                    return DEFAULT_VALUE_WHEN_VERSION_INVALID;
-                }
-
-                final String[] versions = linuxVersion.split("-");
-                return versions.length != 2 ? linuxVersion : versions[1];
-            default:
-                return DEFAULT_VALUE_WHEN_VERSION_INVALID;
-        }
+    public static boolean isJavaWebApp(@NotNull WebApp webApp) {
+        return webApp.javaVersion() != JavaVersion.OFF || StringUtils.containsIgnoreCase(webApp.linuxFxVersion(), "jre8");
```

Comment:
I think it is updated based on the original conditional. Just one thought, is it precise and is there any better choice?

Reply:
Really don't see any better choice from the limited APIs we could leverage. Maybe add the operating system check along with the version check to improve the readability, but not that necessary from the code logic actually.

Reply:
> Maybe add the operating system check along with the version check to improve the readability,

Agree. Better readability leads to less pain in future maintaining work.

Reply:
Cool